core/types/transaction.go

```diff
@@ -1193,11 +1135,44 @@ func (tx *Transaction) AsMessageWithSender(s Signer, baseFee *big.Int, sender *c
 	return msg, err
 }
 
+// CompareFee compares new fee and the tx.Fee() two transactions and returns the output of
+// bigInt Cmp method
+func (tx *Transaction) CompareFee(newFee *big.Int) int {
+	if newFee == nil {
+		return -1 // cannot be compared, so return less than
+	}
+	oldTxFee := tx.GasPrice()
+	return oldTxFee.Cmp(newFee)
+}
+
+func (tx *Transaction) CompareGivenBaseFee(newTx *Transaction, baseFee *big.Int) int {
```

Comment:
What is this function supposed to do?

Reply:
This is not correct, will change this to reflect the logic we had previously

Reply:
Fixed it now, it should technically be comparing gas price first and then if its equal compare the miner tip

Reply:
Don't see a difference

Reply:
Oh true, its not used anymore, deleting it